js/src/components/social/TwitterView.js

```diff
@@ -0,0 +1,106 @@
+import React, { Fragment } from "react";
+import PropTypes from "prop-types";
+import { Alert } from "@yoast/components";
+import { makeOutboundLink } from "@yoast/helpers";
+import { SocialMetadataPreviewForm } from "@yoast/social-metadata-forms";
+import { __, sprintf } from "@wordpress/i18n";
+import styled from "styled-components";
+
+const PremiumInfoText = styled( Alert )`
+	margin-top: 16px;
+ p {
+	 margin: 0;
+ }
+`;
+
+const YoastShortLink = makeOutboundLink();
+
+/* Translators: %s expands to the social medium name, Twitter.  */
+const previewText = sprintf(
+	__(
+		"Do you want to preview what it will look like if people share this post on %s? You can, with", "yoast-components"
+	), "Twitter"
+);
+
+/* Translators: %s expands to Yoast, %s expands to SEO, %s expands to Premium*/
+const upgradeText = sprintf(
+	__(
+		"Find out why you should upgrade to %s %s %s", "yoast-components"
+	), "Yoast", "SEO", "Premium"
+);
+
+/**
+ *
+ * @param {object} props The properties passed to this component.
+ *
+ * @returns {Component} The TwitterView Component.
+ */
+const TwitterView = ( props ) => {
+	const {
+		recommendedReplacementVariables,
+		replacementVariables,
+		description,
+		title,
+		onSelectImageClick,
+		onRemoveImageClick,
+		onDescriptionChange,
+		onTitleChange,
+		imageWarnings,
+		image,
+		isPremium,
+	} = props;
+
+	return (
+		<Fragment>
+			<PremiumInfoText type={ "info" }>
```

Comment:
Could you extract this to a separate component called `SocialPremiumUpsell` or something like that?